test/spec/runners/ssz_static.ex

```diff
@@ -40,47 +78,57 @@ defmodule SszStaticTestRunner do
     expected =
       YamlElixir.read_from_file!(case_dir <> "/value.yaml")
       |> SpecTestUtils.sanitize_yaml()
+      |> sanitize(schema)
 
     %{"root" => expected_root} = YamlElixir.read_from_file!(case_dir <> "/roots.yaml")
     expected_root = expected_root |> SpecTestUtils.sanitize_yaml()
 
     assert_ssz(schema, decompressed, expected, expected_root)
   end
 
-  defp assert_ssz(schema, real_serialized, real_deserialized, expected_root) do
-    {:ok, deserialized} = Ssz.from_ssz(real_serialized, schema)
-    real_deserialized = to_struct_checked(deserialized, real_deserialized)
-
+  defp assert_ssz(schema, real_serialized, real_deserialized, _expected_root) do
+    {:ok, deserialized} = SszEx.decode(real_serialized, schema)
     assert Diff.diff(deserialized, real_deserialized) == :unchanged
-
-    {:ok, serialized} = Ssz.to_ssz(real_deserialized)
+    {:ok, serialized} = SszEx.encode(real_deserialized, schema)
     assert serialized == real_serialized
 
-    root = Ssz.hash_tree_root!(real_deserialized)
-    assert root == expected_root
+    ## TODO Enable when merklelization is enable
+    # root = SszEx.hash_tree_root!(real_deserialized)
+    # assert root == expected_root
   end
 
-  defp to_struct_checked(actual, expected) when is_list(actual) and is_list(expected) do
-    Stream.zip(actual, expected) |> Enum.map(fn {a, e} -> to_struct_checked(a, e) end)
+  defp parse_type(%SpecTestCase{handler: handler}) do
+    Module.concat(Types, handler)
   end
 
-  defp to_struct_checked(vec(_) = actual, vec(_) = expected) do
-    actual
-    |> Aja.Enum.to_list()
-    |> to_struct_checked(Aja.Enum.to_list(expected))
-    |> Aja.Vector.new()
-  end
+  def sanitize(container, module) when is_map(container) do
+    schema = module.schema() |> Map.new()
 
-  defp to_struct_checked(%name{} = actual, %{} = expected) do
-    expected
-    |> Stream.map(fn {k, v} -> {k, to_struct_checked(Map.get(actual, k), v)} end)
-    |> Map.new()
-    |> then(&struct!(name, &1))
+    container
+    |> Enum.map(fn {k, v} -> {k, sanitize(v, Map.fetch!(schema, k))} end)
+    |> then(&struct!(module, &1))
   end
 
-  defp to_struct_checked(_actual, expected), do: expected
+  def sanitize(vector_elements, {:vector, :bool, _size} = _schema), do: vector_elements
 
-  defp parse_type(%SpecTestCase{handler: handler}) do
-    Module.concat(Types, handler)
-  end
+  def sanitize(vector_elements, {:vector, module, _size} = _schema) when is_atom(module),
+    do:
+      vector_elements
+      |> Enum.map(&struct!(module, &1))
```

Comment:
Don't use the pipe operator for just one operation, it makes it less readable. It's easier to just write the following:

```elixir
Enum.map(vector_elements, &struct!(module, &1))
```